notebooks/GA_for_HuggingFace_TF_models.ipynb

```diff
@@ -0,0 +1,541 @@
+    "\n",
+    "#delete the unnecessary models\n",
+    "del HF_model, new_model\n",
+    "\n",
+    "#simply use this GradientAccumulateModel instead of the HF_model\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "STSUeMzCSC6j",
+   "metadata": {
+    "id": "STSUeMzCSC6j"
+   },
+   "source": [
+    "## Example for TFEsmForSequenceClassification"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "03JYV1KUJjQ6",
+   "metadata": {
+    "id": "03JYV1KUJjQ6"
+   },
+   "outputs": [],
+   "source": [
+    "#imports\n",
+    "import pandas as pd\n",
+    "import keras\n",
+    "import tensorflow as tf\n",
+    "import numpy as np\n",
+    "import random\n",
+    "\n",
+    "from gradient_accumulator import GradientAccumulateModel\n",
+    "from tensorflow.keras.layers import Input\n",
+    "from tensorflow.keras.models import Model\n",
+    "\n",
+    "from transformers import AutoTokenizer, TFAutoModelForSequenceClassification, set_seed\n",
+    "from datasets import Dataset\n",
+    "\n",
+    "#reset random state\n",
+    "def set_seeds(s):\n",
+    "    tf.random.set_seed(s)\n",
+    "    tf.config.experimental.enable_op_determinism()\n",
+    "    np.random.seed(s)\n",
+    "    random.seed(s)\n",
+    "    set_seed(s)\n",
+    "\n",
+    "#load a model for sequence regression and the corresponding tokenizer\n",
+    "def load_model(checkpoint,dropout):\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(checkpoint)\n",
+    "    model = TFAutoModelForSequenceClassification.from_pretrained(checkpoint, num_labels=1,classifier_dropout=dropout)\n",
+    "\n",
+    "    return tokenizer,model\n",
+    "\n",
+    "#get training inputs and labels\n",
+    "def load_lists():\n",
+    "    train=pd.read_pickle(\"./hf_example_data.pkl\")\n",
+    "    #reduced size to run on CPU\n",
+    "    train_sequences=train.iloc[:256,0].tolist() \n",
+    "    train_labels=train.iloc[:256,1].tolist()\n",
+    "\n",
+    "    return train_sequences, train_labels\n",
+    "\n",
+    "#create a HuggingFace Dataset\n",
+    "def create_dataset(tokenizer,seqs,labels):\n",
+    "    #reduced sequence length to run on CPU\n",
+    "    tokenized = tokenizer(seqs,max_length=64, padding=True, truncation=True) \n",
+    "    dataset = Dataset.from_dict(tokenized)\n",
+    "    dataset = dataset.add_column(\"labels\", labels)\n",
+    "\n",
+    "    return dataset\n",
+    "\n",
+    "#create a tf dataset\n",
+    "def tf_dataset(model,tokenizer,dataset,batch,seed,shuffle=True):\n",
+    "    tf.random.set_seed(seed)     \n",
+    "    tf_set = model.prepare_tf_dataset(\n",
+    "      dataset,\n",
+    "      batch_size=batch,\n",
+    "      shuffle=shuffle,\n",
+    "      tokenizer=tokenizer)\n",
+    "\n",
+    "    return tf_set\n",
+    "\n",
+    "#training function\n",
+    "def main(checkpoint,dropout=0.2,batch=8,accum=1,epoch=10,lr=2e-5,seed=42,GA=True):\n",
```

Comment:
Could you rename this function to `run_experiment` or similar, and add spaces between the function arguments, similarly as done before?

Reply:
> Could you remove the *.pkl file, as I assume it is generated by the program and not needed to run this notebook, @0syrys?

Its actually the input data for the experiment


> Could you rename this function to `run_experiment` or similar, and add spaces between the function arguments, similarly as done before?

Done

Reply:
Great! Good job :)